hack/generate-operator-bundle.py

```diff
@@ -4,88 +4,172 @@
-    sys.exit(1)
-
-outdir = sys.argv[1]
-prev_version = sys.argv[2]
-git_num_commits = sys.argv[3]
-git_hash = sys.argv[4]
-hive_image = sys.argv[5]
-
-full_version = "%s.%s-sha%s" % (VERSION_BASE, git_num_commits, git_hash)
-print("Generating CSV for version: %s" % full_version)
-
-if not os.path.exists(outdir):
-    os.mkdir(outdir)
-
-version_dir = os.path.join(outdir, full_version)
-if not os.path.exists(version_dir):
-    os.mkdir(version_dir)
-
-# Copy all CSV files over to the bundle output dir:
-crd_files = os.listdir('config/crds/')
-for file_name in crd_files:
-    full_path = os.path.join('config/crds', file_name)
-    if (os.path.isfile(os.path.join('config/crds', file_name))):
-        shutil.copy(full_path, os.path.join(version_dir, file_name))
-
-with open('config/templates/hive-csv-template.yaml', 'r') as stream:
-    csv = yaml.load(stream, Loader=yaml.SafeLoader)
-
-csv['spec']['install']['spec']['clusterPermissions'] = []
-
-# Add our operator role to the CSV:
-with open('config/operator/operator_role.yaml', 'r') as stream:
-    operator_role = yaml.load(stream, Loader=yaml.SafeLoader)
-    csv['spec']['install']['spec']['clusterPermissions'].append(
-        {
-            'rules': operator_role['rules'],
-            'serviceAccountName': 'hive-operator',
-        })
-
-# Add our deployment spec for the hive operator:
-with open('config/operator/operator_deployment.yaml', 'r') as stream:
-    operator_components = []
-    operator = yaml.load_all(stream, Loader=yaml.SafeLoader)
-    for doc in operator:
-        operator_components.append(doc)
-    operator_deployment = operator_components[1]
-    csv['spec']['install']['spec']['deployments'][0]['spec'] = operator_deployment['spec']
-
-# Update the deployment to use the defined image:
-csv['spec']['install']['spec']['deployments'][0]['spec']['template']['spec']['containers'][0]['image'] = hive_image
-
-# Update the versions to include git hash:
-csv['metadata']['name'] = "hive-operator.v%s" % full_version
-csv['spec']['version'] = full_version
-csv['spec']['replaces'] = "hive-operator.v%s" % prev_version
-
-# Set the CSV createdAt annotation:
-now = datetime.datetime.now()
-csv['metadata']['annotations']['createdAt'] = now.strftime("%Y-%m-%dT%H:%M:%SZ")
-
-# Write the CSV to disk:
-csv_filename = "hive-operator.v%s.clusterserviceversion.yaml" % full_version
-csv_file = os.path.join(version_dir, csv_filename)
-with open(csv_file, 'w') as outfile:
-    yaml.dump(csv, outfile, default_flow_style=False)
-print("Wrote ClusterServiceVersion: %s" % csv_file)
 
+OSD_HIVE_IMAGE_DEFAULT = 'quay.io/app-sre/hive'
+OPERATORHUB_HIVE_IMAGE_DEFAULT = 'quay.io/openshift-hive/hive'
+CHANNEL_DEFAULT = 'alpha'
+BUNDLE_DIR = 'bundle'
+PACKAGE_FILE = os.path.join(BUNDLE_DIR, 'hive.package.yaml')
+
+
+def current_operatorhub_version():
+    result = subprocess.check_output("git describe --abbrev=0", shell=True).strip()
```

Comment:
`check_output` will return a `bytes` object, I think it would be safer to decode it so you don't have problems later on.

```
result = subprocess.check_output("git describe --abbrev=0", shell=True).strip().decode()
```

There is a similar line below

Reply:
oh, I just realized this is executed by `py2.7` so what I said is inaccurate. Please ignore this.